kong/cli/services/nginx.lua

```diff
@@ -59,11 +59,23 @@ local function get_current_user()
   return IO.os_execute("whoami")
 end
 
+local function get_primary_group(user)
+  return IO.os_execute("id -g -n "..user)
+end
+
+local function is_root()
+  local _, exit_code = IO.os_execute("[[ $EUID -eq 0 ]]")
+  return exit_code == 0
+end
+
 local function prepare_nginx_configuration(configuration, ssl_config)
+
+  local current_user = get_current_user()
+
   -- Extract nginx config from kong config, replace any needed value
   local nginx_config = configuration.nginx
   local nginx_inject = {
-    user = get_current_user(),
+    user = is_root() and "user "..current_user.." "..get_primary_group(current_user)..";" or "",
```

Comment:
this will not work for any user that is not root, which is limiting the ability to run Kong process as any user, specifically for security purposes a sysadmin would want to silo the kong execution into a specific user, this forbids that, its too dangerous to rely only on root users.